Adding 1D Flightdistance tagging variables from cmssw 8_0_0_pre6

IHEP-CMS opened the pull request:
Dear Experts,
                                   An update for the tagging variables has been made after implementing flightdistance1d as well. Please do have a look and merge if the things seems fine.

Cheers,
Sarmad Masood Shaheen.

smshaheen committed:
Updated files after inclusion of flightDist1d

smshaheen committed:
Removing the AbsflightDist from TaggingVariable

cmsbuild commented:
A new Pull Request was created by  @IHEP-CMS  for CMSSW_8_1_X.

It involves the following packages:

DataFormats/BTauReco
RecoBTag/SecondaryVertex


@cmsbuild, @cvuosalo, @davidlange6, @slava77 can you please review it and eventually sign? Thanks.
@ferencek, @acaudron, @pvmulder, @imarches this is something you requested to watch as well.
@slava77, @degano, @smuzaffar you are the release manager for this.

cms-bot commands are list here cms-sw/cmssw#13028

slava77 commented:
@cmsbuild please test

@IHEP-CMS 
- please change the PR title/subject to reflect that 1D flight distance tagging variables are added
- in the PR description (the first block that starts now with "Dear experts") please add a brief summary about changes in physics performance, if any; and a link to slides in a POG meeting.

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
-1 
 
Tested at: e95bebfde1fe88fa1d8b4bb450fcb1fe62473bae
I found errors in the following unit tests: 
 
 ---> test runtestTqafTopEventSelection had ERRORS
 
 
you can see the results of the tests here: 
 https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-13419/11432/summary.html

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-13419/11432/summary.html

ferencek commented on DataFormats/BTauReco/interface/TaggingVariable.h:
```diff
@@ -78,7 +78,9 @@ namespace reco {
       vertexJetDeltaR,                          // pseudoangular distance between jet axis and secondary vertex direction
-
+      
+      flightDistance1dVal,                      // Linear distance between primary and secondary vertex
```
"linear distance" is probably not the best description. Maybe "longitudinal distance" or "longitudinal distance along the z axis"...

IHEP-CMS replied:
Thanks for a nice suggestion. The related changes has been made.

slava77 commented:
@ferencek @pvmulder 
Please clarify if this was already presented in the BTV meetings.

imarches commented:
@slava77 yes, this PR was coordinated with BTV. I will remember to Sarmad to fulfill your request

IHEP-CMS committed:
Changing the description of the 1D flightDist

The description of 1D flighDist has been changed from:

Linear distance -> Longitudinal distance along z-axis.

cmsbuild commented:
Pull request #13419 was updated. @cmsbuild, @cvuosalo, @davidlange6, @slava77 can you please check and sign again.

IHEP-CMS committed:
Changing the description of the 1D flightDist

The description of 1D flighDist has been changed from:

Linear distance -> Longitudinal distance along z-axis.

cmsbuild commented:
Pull request #13419 was updated. @cmsbuild, @cvuosalo, @davidlange6, @slava77 can you please check and sign again.

slava77 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
-1 
 
Tested at: 51314c125c4eb5e2a9c5add5ff7838e3ab2c4089
I found errors in the following unit tests: 
 
 ---> test runtestTqafTopEventSelection had ERRORS
 
 
you can see the results of the tests here: 
 https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-13419/11486/summary.html

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-13419/11486/summary.html

slava77 commented:
There is a segfault in the unit tests is present in the IB as well.
```
Failure using ttSemiLepSignalSelMVATrainTreeSaver_cfg.py: status 139
```
So, this is not related to this PR.

slava77 commented on DataFormats/BTauReco/interface/TemplatedSecondaryVertexTagInfo.h:
```diff
@@ -58,7 +58,7 @@ class TemplatedSecondaryVertexTagInfo : public BaseTagInfo {
+                Measurement1D                   dist1d,dist2d, dist3d;
                 GlobalVector                    direction;
 		
 		// Used by ROOT storage
```
I suppose, the class version has to be updated.

@Dr15Jones  do we have a method to require a version change in a templated type after the data members change

Dr15Jones replied:
The version does need to be updated.

I'm not sure if putting the checksums for each template instantiation into the appropriate classes_def.xml would catch the problem or not.

IHEP-CMS replied:
Dear Salva, 
                        I agree the class version need to be updated after change the template thus may need to include an updated checksum. Please do let me know if I have to update classes_def.xml file as well.

slava77 replied:
Hi Sarmad,
Please update the version in this file (from 11 to 12 in two places).
I will need some instructions from Chris on how to update classes_def.xml with checksums included.
If I understand correctly, the change in classes_def.xml is not required at this point.

IHEP-CMS replied:
In classes_def.xml file right?
Also there may be some older version which should not be replaced. I mean I will keep the other version as it is but will add version 12 for Measurement1D. Please do correct me if I am wrong.

slava77 replied:
as I noticed in the previous comment, the class version should be updated in this file
(TemplatedSecondaryVertexTagInfo.h)

slava77 commented:
After fixing the class version locally, it is now possible to read old and new files.

Here is an example of  pfInclusiveSecondaryVertexFinderTagInfos in ttbar (no PU) workflow 25.0 [in these plots red color is for the baseline: values there come out at zero as expected; the new is in the black histogram and shows something sensible)

dist1d.value
![all_origvstest13419-pass-bdf83c6_ttbarwf25p0c_recocandidateedmptrsrecojettaginforecoiptaginforecovertexcompositeptrcandidaterecotemplatedsecondaryvertextaginfos_pfinclusivesecondaryvertexfindertaginfos__reco_obj_m_svdata_dist1d_value](https://cloud.githubusercontent.com/assets/4676718/13300694/3b65e480-daf6-11e5-9d78-8a335a826668.png)

dist1d.error
![all_origvstest13419-pass-bdf83c6_ttbarwf25p0c_recocandidateedmptrsrecojettaginforecoiptaginforecovertexcompositeptrcandidaterecotemplatedsecondaryvertextaginfos_pfinclusivesecondaryvertexfindertaginfos__reco_obj_m_svdata_dist1d_error](https://cloud.githubusercontent.com/assets/4676718/13300709/4749a03e-daf6-11e5-9ee4-12877e38fbd0.png)

IHEP-CMS committed:
Updating Class_version to 12

in the file:
TemplatedSecondaryVertexTagInfo.h
The class version has been updated from 11 to 12.

cmsbuild commented:
Pull request #13419 was updated. @cmsbuild, @cvuosalo, @davidlange6, @slava77 can you please check and sign again.

slava77 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

slava77 commented:
+1

for https://github.com/cms-sw/cmssw/pull/13419	8d054e1
- code changes look ok, in line with the description: dist1d variable is added to the tag infos
- locally run tests pass and comparisons with baseline show differences only in dist1d distributions which are now filled (see plots earlier)


jenkins tests apparently got stuck.
The previous iteration of jenkins tests showed no differences in DQM comparisons; there were false-positive differences in ZEEFS plots using fwlite-based script, but these were affected by inappropriate class version resolution.

slava77 commented:
@cmsbuild please test
it looks like the last attempt got stuck  (https://cmssdt.cern.ch/jenkins/job/ib-any-integration/11512/console the console is showing no activity since somewhere in the middle of the code checkout process)

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next CMSSW_8_1_X IBs after it passes the integration tests. This pull request requires discussion in the ORP meeting before it's merged. @slava77, @davidlange6, @degano, @smuzaffar

cmsbuild commented:
-1 
 
Tested at: 8d054e14b324bfd7c19c35551fe22b85a8cc87b7
I found errors in the following unit tests: 
 
 ---> test runtestTqafTopEventSelection had ERRORS
 
 
you can see the results of the tests here: 
 https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-13419/11534/summary.html
